cache/class-vary-cache.php

```diff
@@ -211,14 +212,17 @@ private static function decrypt_cookie_value( $cookie_value ) {
 	 * @access  private
 	 */
 	private static function parse_group_cookie() {
-		if ( isset( $_COOKIE[ self::COOKIE_SEGMENT ] ) ) {
-			$cookie_value = sanitize_text_field( wp_unslash( $_COOKIE[ self::COOKIE_SEGMENT ] ) );
+		if ( isset( $_COOKIE[ self::COOKIE_SEGMENT ] ) || isset( $_COOKIE[ self::COOKIE_AUTH ] ) ) {
+
 			if ( self::is_encryption_enabled() ) {
+				$cookie_value = str_replace( self::VERSION_PREFIX, '', $_COOKIE[ self::COOKIE_AUTH ] );
```

Comment:
This line should be moved after the decryption. Varnish isn't aware of it so the version will need to be part of the encrypted content.

Reply:
Yes, this was an untested WIP :)